app/src/main/java/org/fossasia/openevent/app/data/auth/EncryptionServiceImpl.java

```diff
@@ -0,0 +1,215 @@
+package org.fossasia.openevent.app.data.auth;
+
+import android.content.Context;
+import android.os.Build;
+import android.security.KeyPairGeneratorSpec;
+import android.support.annotation.RequiresApi;
+import android.util.Base64;
+import android.util.Log;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.inject.Inject;
+import javax.security.auth.x500.X500Principal;
+
+import timber.log.Timber;
+
+public class EncryptionServiceImpl implements EncryptionService {
+
+    private Context context;
+    private static final String KEYSTORE = "AndroidKeyStore";
+    private static final String ALIAS = "MY_APP";
+    private static final String TYPE_RSA = "RSA";
+    private static final String CYPHER = "RSA/ECB/PKCS1Padding";
+    private static final String ENCODING = "UTF-8";
+
+    @Inject
+    public EncryptionServiceImpl(Context context) {
+        this.context = context;
+    }
+
+    @Override
+    public String encrypt(String credential) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
```

Comment:
We are targetting above JB, so why this check?

Reply:
We have a min API level of 16 and this requires API Level 18

Reply:
So, on versions below 18, it will not encrypt the String, this is a very big security issue

Reply:
The **KeyStoreGeneratorSpec** which is used to generate and stores it using android keystore was introduced in API level 18 only. To handle cases before that we can encrypt it normally without having the need to use KEYS. Should I use that for this case?

Reply:
Yes